includes/data-stores/class-wc-order-data-store-cpt.php

```diff
@@ -772,15 +775,33 @@ protected function get_wp_query_args( $query_vars ) {
 			}
 		}
 
-		// Add the 'wc-' prefix to status if needed.
 		if ( ! empty( $query_vars['post_status'] ) ) {
+			// Add the 'wc-' prefix to status if needed.
 			if ( is_array( $query_vars['post_status'] ) ) {
 				foreach ( $query_vars['post_status'] as &$status ) {
 					$status = wc_is_order_status( 'wc-' . $status ) ? 'wc-' . $status : $status;
 				}
 			} else {
 				$query_vars['post_status'] = wc_is_order_status( 'wc-' . $query_vars['post_status'] ) ? 'wc-' . $query_vars['post_status'] : $query_vars['post_status'];
 			}
+
+			// Verify post_status is actually registered.
+			$post_stati = is_array( $query_vars['post_status'] ) ? $query_vars['post_status'] : array( $query_vars['post_status'] );
+			foreach ( $post_stati as $post_status ) {
+				$post_status_exists = get_post_stati( array( 'name' => $post_status ) );
+				if ( empty( $post_status_exists ) ) {
+					$message = sprintf(
+						/* translators: %s: status name */
+						__(
+							'The provided order query contains an order status that is not registered as a custom post status (status provided is %s). It is possible there is something removing an expected registered custom post status before the query is made.',
```

Comment:
Perhaps second part of the error message is not helpful, because [unregistered post statuses are definately removed in WP query](https://github.com/WordPress/WordPress/blob/b66ff2f68daf3a6c83154f2dd841c9a5dedd4c4e/wp-includes/class-wp-query.php#L2465).